data-model/src/main/scala/za/co/absa/enceladus/model/MappingTable.scala

```diff
@@ -43,7 +43,7 @@ case class MappingTable(name: String,
                         dateDisabled: Option[ZonedDateTime] = None,
                         userDisabled: Option[String] = None,
 
-                        locked: Boolean = false,
+                        locked: Option[Boolean] = None,
```

Comment:
Using the option wrapper makes sense to differentiate the unset situation by `None`. However, what I find less ideal is the fact, that now you have to default-to-false on empty and you are now doing it in many places.

Some places call `locked.getOrElse(false)`, other places `forall(_.locked.contains(true)` etc. - so the defaulting to `false` is spread around (and god-forbid you ever had to change it and forgot one place).

I think it would be better to do the defaulting once and for all - e.g. by providing:
```scala
class Dataset {
  def isLocked: Boolean = locked.getOrElse(false)
}
```

That way, all other calls would never have to default themselves.

Reply:
This does make sense, however if I add that method to the `VersionedModel`, it looks like serializing and deserializing doesn't work the same anymore and on every update the value is added as false, which I'm not sure if it's what we want.

Reply:
Perhaps the method could be excepted from being serialized (some annotation perhaps?)

Reply:
So, I have tried what I suggested and, truly, integration tests fail on SerDe-related errors. The nature of the errors, however, suggests that the method `isLocked` is used as a getter for the SerDe processing and that's why the tests fail.

I have also tried the same, just with the method bearing a different name (e.g. `lockedWithDefault`) and the problem disappeared. So, I believe the idea can be used, just rename the method :D

Reply:
Tried `lockedWithDefault`

Reply:
I must admit, I am not really fond of the function name `lockedWithDefault`. I don't have much better idea though.
`lockedDefined`?
Or a way I remember from Delphi, where the field has an _F_ prefix and getter/setter was without it. What do you think about this style:
```scala
    val fLocked: Option[Boolean]
    def Locked: Boolean = fLocked.getOrElse(false)
```

Reply:
Yeah, the f-prefix has been rejected on the CQC meeting, so it's of the table. 😉 
And I can live with `lockedWithDefault`, so up to @AdrianOlosutean if he keeps it or replaces it 😉